akka-actor-typed/src/main/scala/akka/actor/typed/internal/Supervision.scala

```diff
@@ -317,18 +318,28 @@ private class RestartSupervisor[O, T, Thr <: Throwable: ClassTag](initial: Behav
       case _: Restart ⇒
     }
 
-    try {
-      val newBehavior = Behavior.validateAsInitial(Behavior.start(initial, ctx.asInstanceOf[TypedActorContext[T]]))
-      val nextBehavior = restartingInProgress match {
-        case OptionVal.None ⇒ newBehavior
-        case OptionVal.Some((stashBuffer, _)) ⇒
-          restartingInProgress = OptionVal.None
-          stashBuffer.unstashAll(ctx.asScala.asInstanceOf[scaladsl.ActorContext[Any]], newBehavior.unsafeCast)
+    // can throw, unrelated to unstashAll, keep separate
+    val newBehavior =
+      try Behavior.validateAsInitial(Behavior.start(initial, ctx.asInstanceOf[TypedActorContext[T]])) catch {
+        case NonFatal(e) ⇒ Behavior.failed[T](e)
```

Comment:
Because `Behavior.validateAsInitial` can raise its own exception, unrelated... lets be sure that if we capture a failure in unstashAll, that it is just from that.